### DISCUSSION TITLE
How can I query the cached status of a task?

### DISCUSSION THREAD

**Discussion by chadrik** [moonrepo/moon#1759] [Q&A]
Is there a way to query whether or not the inputs of a task are fully cached, and thus whether the task will be skipped?  

I found a few apparent dead-ends:

`moon query tasks --json` returns a `state` field for each task, but it doesn't seem to reflect stale inputs.

---

`moon query hash` seems like it could play a role in a solution, but I don't know how to get the hash of the inputs for a task. 

---

`moon query touched-files` is not quite what I want because the cache state and the VCS state are not necessarily the same.

--- 

What would be really useful would be a `--dry` flag for `moon run`, e.g.:

```
moon run --dry --summary
```
---

Any help would be greatly appreciated!

**Comment by milesj** [comment]
There's currently no way to access that information without running the task, as it's coupled to the task runner.
